File: DemoAppObj-C/YahooSearchKitDemo/View Controllers/YSKViewController.m
//
//  YSKViewController.m
//  YahooSearchKitDemo
//
//  Copyright 2015 Yahoo! Inc.
//  Licensed under the terms of the zLib license. See LICENSE file at the root of this project for license terms.
//

#import "YSKViewController.h"

@interface YSKViewController () <YSLSearchViewControllerDelegate>

@end

@implementation YSKViewController

- (void)presentViewController:(UIViewController *)viewControllerToPresent animated:(BOOL)flag completion:(void (^)(void))completion {
    UIApplication.sharedApplication.statusBarStyle = UIStatusBarStyleDefault;
    [super presentViewController:viewControllerToPresent animated:flag completion:completion];
}

- (void)searchViewControllerDidTapLeftButton:(YSLSearchViewController*)searchViewController {
    UIApplication.sharedApplication.statusBarStyle = UIStatusBarStyleLightContent;
}

#pragma mark - YSLSearchViewControllerDelegate methods

- (YSLQueryAction)searchViewController:(YSLSearchViewController *)searchViewController actionForQueryString:(NSString *)queryString {
    return YSLQueryActionSearch;
}

@end

File: DemoAppSwift/YahooSearchKitDemo/View Controllers/YSKViewController.swift
//
//  YSKViewController.swift
//  YahooSearchKitDemo
//
//  Copyright 2015 Yahoo! Inc.
//  Licensed under the terms of the zLib license. See LICENSE file at the root of this project for license terms.
//

import UIKit

class YSKViewController: UIViewController, YSLSearchViewControllerDelegate {

    override func presentViewController(viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)?) {
        UIApplication.sharedApplication().statusBarStyle = .Default
        super.presentViewController(viewControllerToPresent, animated: flag, completion: completion)
    }

    // MARK: YSLSearchViewControllerDelegate methods
    func searchViewControllerDidTapLeftButton(searchViewController: YSLSearchViewController!) {
        UIApplication.sharedApplication().statusBarStyle = .LightContent
    }
    
    func searchViewController(searchViewController: YSLSearchViewController!, actionForQueryString queryString: String!) -> YSLQueryAction {
        return YSLQueryAction.Search
    }
}
